src/components/form/sw-select/sw-select.vue

```diff
@@ -83,25 +128,164 @@ export default {
   },
 
   props: {
+    /**
+     * An array of objects with the labelProperty and valueProperty.
+     *
+     * @example [{label: 'Option A', value: 'a'}, {label: 'Option B', value: 'b'}]
+     */
+    options: {
+      type: Array,
+      required: true,
+    },
+
+    /**
+     * Toggles if either one or more items can be selected.
+     */
+    enableMultiSelection: {
+      type: Boolean,
+      default: false,
+    },
+
+    /**
+     * Dependent on multiSelection, either a single value or an array of values.
+     */
     value: {
-      type: String,
+      type: [String, Number, Boolean, Array, null, undefined],
       required: false,
       default: null,
     },
 
+    /**
+     * The object key of the label property.
+     */
+    labelProperty: {
+      type: String,
+      required: false,
+      default: 'label',
+    },
+
+    /**
+     * The object key to use for the value.
+     */
+    valueProperty: {
+      type: String,
+      required: false,
+      default: 'value',
+    },
+
+    /**
+     * The number of items that are expanded by default.
+     */
+    valueLimit: {
+      type: Number,
+      required: false,
+      default: 5,
+    },
+
+    /**
+     * The label for the select field itself.
+     */
+    label: {
+      type: String,
+      required: true,
+    },
+
+    /**
+     * The placeholder for the select field.
+     */
     placeholder: {
       type: String,
       required: false,
-      default: null,
+      default: '',
     },
 
-    options: {
-      type: Array,
+    /**
+     * Toggles the loading state of the select field.
+     */
+    isLoading: {
+      type: Boolean,
+      required: false,
+      default: false,
+    },
+
+    /**
+     * Disables or enables the select field.
+     */
+    disabled: {
+      type: Boolean,
+      required: false,
+      default: false,
+    },
+
+    /**
+     * Toggles a button to clear all selections.
+     */
+    showClearableButton: {
```

Comment:
I would prefer to change it to `hideClearableButton`. And the clearable button should be visible by default. But this should decide @Weltraumakustik what the default behavior should look like.